### PR TITLE
feat: event emitter will now emit new_txid in-case a tx replaces another

### PR DESCRIPTION
**Pull request by usagi32**
This extra information is required by an issue in the blockchain-api [here ](https://github.com/hirosystems/stacks-blockchain-api/pull/2141)and [here](https://github.com/hirosystems/stacks-blockchain-api/issues/1813).

Event emitter will now give to the api node which transaction replaced another.

### CODE REVIEW

**Review by jferrant — APPROVED** [review]

**Comment by jferrant** [comment]
I could be wrong, but I think this may need to be opened against develop, not master.

**Comment by usagi32** [comment]
@jferrant yes didn't know it existed.

**Review by jcnelson — APPROVED** [review]
This LGTM, pending CI passes.  Thanks @usagi32!

**Comment by hstove** [comment]
@usagi32 you'll have to fix some of the formatting errors before CI runs tests - you can see these by running `cargo fmt-stacks`

**Review by obycode — DISMISSED** [review]
LGTM!

**Review by jcnelson — DISMISSED** [review]
LGTM. Thanks @usagi32!

**Review by hstove — DISMISSED** [review]

**Comment by hstove** [comment]
Ah, unfortunately there's a new compilation error, assuming from after merging in develop.

**Comment by jbencin** [comment]
Fixed build errors and merged latest `develop`

**Review by hstove — DISMISSED** [review]

**Review by obycode — APPROVED** [review]
LGTM!

**Review by obycode — CHANGES_REQUESTED** [review]
Could we add a changelog entry for this?

**Review by hstove — APPROVED** [review]

File: CHANGELOG.md
Comment (obycode) [inline CHANGELOG.md:10]: Can you moved this under "Changed"?

**Review by obycode — DISMISSED** [review]
LGTM!

**Comment by obycode** [comment]
Oh, it's not running the CI jobs because the last commit only changed a markdown file. Looks like we need some adjustments to the CI rules.

**Comment by hstove** [comment]
@obycode I think it's something to do with me pushing to this forked branch, I'm not sure. Maybe I'll need to open a new PR...

**Comment by obycode** [comment]
Looks like merging develop was enough to get it to trigger, so maybe my Markdown theory was correct?

**Review by obycode — APPROVED** [review]

**Comment by hstove** [comment]
Closing, reponed as https://github.com/stacks-network/stacks-core/pull/5605

**Pull request closed without merging**
